go/dpagg/count_test.go

```diff
@@ -345,6 +345,93 @@ func TestCountThresholdedResult(t *testing.T) {
+	}{
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+			want:    noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+		},
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: -5, UpperBound: 5},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: 5},
+		},
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: -5, UpperBound: -5},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: 0},
+		},
+		// Infinite bounds happens for extremely small alpha for Gaussian.
+		{
+			confInt: noise.ConfidenceInterval{LowerBound: math.Inf(-1), UpperBound: math.Inf(1)},
+			want:    noise.ConfidenceInterval{LowerBound: 0, UpperBound: math.Inf(1)},
+		},
+	} {
+		c := getNoiselessCount()
+		// This makes Noise interface return the pre-processing confidence interval when ComputeConfidenceIntervalInt64 is called.
+		c.noise = getMockConfInt(tc.confInt)
+
+		c.Result()
+		got, _ := c.ComputeConfidenceInterval(0.1) // alpha is ignored in mockConfInt.
+
+		if got.LowerBound != tc.want.LowerBound {
+			t.Errorf("TestCountComputeConfidenceIntervalPostProcessing(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, LowerBounds are not equal",
+				tc.confInt.LowerBound, tc.confInt.UpperBound, got.LowerBound, tc.want.LowerBound)
+		}
+		if got.UpperBound != tc.want.UpperBound {
+			t.Errorf("TestCountComputeConfidenceIntervalPostProcessing(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, UpperBounds are not equal",
+				tc.confInt.LowerBound, tc.confInt.UpperBound, got.UpperBound, tc.want.UpperBound)
+		}
+	}
+}
+
+func TestCountComputeConfidenceIntervalComputation(t *testing.T) {
+	// Tests returned confidence intervals using Laplace or Gaussian for a given count.
+	for _, tc := range []struct {
+		desc string
+		opt  *CountOptions
+		want noise.ConfidenceInterval
+	}{
+		{
+			desc: "Gaussian",
+			opt:  &CountOptions{Epsilon: 0.1, Delta: 0.1, Noise: getNoiselessConfInt(noise.Gaussian())},
+			want: noise.ConfidenceInterval{LowerBound: 8, UpperBound: 12},
+		},
+		{
+			desc: "Laplace",
+			opt:  &CountOptions{Epsilon: 0.1, Noise: getNoiselessConfInt(noise.Laplace())},
+			want: noise.ConfidenceInterval{LowerBound: 3, UpperBound: 17},
+		},
+	} {
+		c := NewCount(tc.opt)
+		for i := 0; i < 10; i++ {
+			c.Increment()
+		}
+		c.Result()
+		got, _ := c.ComputeConfidenceInterval(0.5)
+
+		if got.LowerBound != tc.want.LowerBound {
+			t.Errorf("TestCountComputeConfidenceIntervalComputation(Noise: %s)=%0.10f, want %0.10f, LowerBounds are not equal",
+				tc.desc, got.LowerBound, tc.want.LowerBound)
+		}
+		if got.UpperBound != tc.want.UpperBound {
+			t.Errorf("TestCountComputeConfidenceIntervalComputation(Noise: %s)=%0.10f, want %0.10f, UpperBounds are not equal",
+				tc.desc, got.UpperBound, tc.want.UpperBound)
+		}
+	}
+}
+
+func TestCountComputeConfIntCannotBeCalledBeforeResult(t *testing.T) {
+	c := getNoiselessCount()
+	// Calling ComputeConfidenceInterval without calling Result() first will produce an error.
```

Comment:
Again, please move this above the function signature: `Tests that calling COmputeConfidenceInterval without calling Result() produces an error.`